### ISSUE TITLE
(Bug report) Trilium generates errors on startup, unable to recover

### ISSUE THREAD

**Issue by davidbrownell** [zadam/trilium#2901]
### Trilium Version

0.51.2

### What operating system are you using?

Windows

### What is your setup?

Local + server sync

### Operating System Version

Windows 11

### Description

I receive the CKEditorError errors below (both in the electron app and browser) which block Trilium from starting - I am unable to use the UX at all (making it very difficult to debug what might be happening :) ).

Does Trilium have a safe mode or similar functionality that I can use to try to determine which note is causing the problem?

Thanks!

``` 
10:04:51 Uncaught error: Message: Uncaught CKEditorError: Cannot read properties of null (reading 'getChildren')
Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-Cannot read properties of null (reading 'getChildren'), URL: http://localhost:3333/#root/rD4BhWEW86Sv/YIrxaQCW2otf/FOPJAM6MuqU5/pDBHy8FhuNGS-bZD9, Line: 6, Column: 350334, Error object: {"name":"CKEditorError","context":{}}, Stack: TypeError: Cannot read properties of null (reading 'getChildren')
    at np._handleContainerChildrenMutations (<anonymous>:6:784465)
    at np.handle (<anonymous>:6:783814)
    at Object.<anonymous> (<anonymous>:6:787335)
    at Object.fire (<anonymous>:6:351884)
    at Ji (<anonymous>:6:420026)
    at Xi.fire (<anonymous>:6:419526)
    at js._onMutations (<anonymous>:6:471332)
logError @ ws.js:17
ckeditorerror.js:104 
        
       Uncaught TypeError: Cannot read properties of null (reading 'getChildren')
    at np._handleContainerChildrenMutations (<anonymous>:6:784465)
    at np.handle (<anonymous>:6:783814)
    at Object.<anonymous> (<anonymous>:6:787335)
    at Object.fire (<anonymous>:6:351884)
    at Ji (<anonymous>:6:420026)
    at Xi.fire (<anonymous>:6:419526)
    at js._onMutations (<anonymous>:6:471332)
ws.js:17 
        
       10:04:51 Uncaught error: Message: Uncaught TypeError: Cannot read properties of undefined (reading 'clientWidth'), URL: http://localhost:3333/app/widgets/tab_row.js, Line: 313, Column: 58, Error object: {}, Stack: TypeError: Cannot read properties of undefined (reading 'clientWidth')
    at get tabWidths [as tabWidths] (http://localhost:3333/app/widgets/tab_row.js:313:58)
    at TabRowWidget.layoutTabs (http://localhost:3333/app/widgets/tab_row.js:359:41)
    at ResizeObserver.resizeListener (http://localhost:3333/app/widgets/tab_row.js:284:18)
logError @ ws.js:17
tab_row.js:313 
        
       Uncaught TypeError: Cannot read properties of undefined (reading 'clientWidth')
    at get tabWidths [as tabWidths] (tab_row.js:313:58)
    at TabRowWidget.layoutTabs (tab_row.js:359:41)
    at ResizeObserver.resizeListener (tab_row.js:284:18)
jquery.min.js:2 
        
       Uncaught (in promise) DOMException: Failed to execute 'appendChild' on 'Node': The new child element contains the parent.
    at HTMLTableSectionElement.<anonymous> (http://localhost:3333/libraries/jquery.min.js:2:49823)
    at He (http://localhost:3333/libraries/jquery.min.js:2:48091)
    at S.fn.init.append (http://localhost:3333/libraries/jquery.min.js:2:49724)
    at S.fn.init.tableSort (eval at <anonymous> (http://localhost:3333/app/services/bundle.js:18:20), <anonymous>:13:3919)
    at HTMLTableElement.eval (eval at <anonymous> (http://localhost:3333/app/services/bundle.js:18:20), <anonymous>:13:6121)
    at Function.each (http://localhost:3333/libraries/jquery.min.js:2:3003)
    at S.fn.init.each (http://localhost:3333/libraries/jquery.min.js:2:1481)
    at o.fn.fancyTable (eval at <anonymous> (http://localhost:3333/app/services/bundle.js:18:20), <anonymous>:13:3936)
    at exports.CreateTabularDataElement (eval at <anonymous> (http://localhost:3333/app/services/bundle.js:18:20), <anonymous>:285:12)
    at eval (eval at <anonymous> (http://localhost:3333/app/services/bundle.js:18:20), <anonymous>:530:52)
```

**Comment by zadam** [comment]
Hi, yes, there's a safe mode in case custom scripts/widgets misbehave and crash the app (there's no sandboxing):

https://github.com/zadam/trilium/wiki/Troubleshooting#broken-script-prevents-application-startup

> In such cases you can start Trilium in "safe mode" which will not execute any custom scripts:

```
TRILIUM_SAFE_MODE=true ./trilium
```

> Depending on the Trilium distribution, there are also ready made scripts for this: trilium-safe-mode.bat and trilium-safe-mode.sh.

**Comment by davidbrownell** [comment]
Thanks for the quick reply! Launching in safe mode helped me to see the error `SqliteError: database disk image is malformed`, which explains the behavior.

Unfortunately, the weekly backup was corrupt as well so I had to restore to the monthly backup. It seems that my server is in a corrupt state as well.

A couple of feature requests based on this experience:
- Trilium should surface these problems immediately when they are encountered so the user can take action. My hunch is this started to happen for me yesterday. Ideally, I would have fixed the problem then.
- Trilium should not make backups of malformed databases.
- Trilium should not sync with a malformed backend.

Thanks for the great product!

**Comment by davidbrownell** [comment]
Actually, this may not be the root cause of the problem. I am digging a bit more to try to figure out where things are going off the rails and will update this thread in a couple of hours when I have more info.

**Comment by davidbrownell** [comment]
TL;DR: There is not a bug with Trilium, however there were bugs with my code and with Chrome/Edge.

**More info:**
Please disregard my comments about syncing above, as the database corruption that I had experienced was the result of my attempts to work around the hang (before I knew about safe mode) by opening the SQLite db and manually setting the delete flag on what I thought to be the problematic notes.

This was a really bad idea in hindsight. It was a really bad idea at the time, but I unfortunately ignored my instincts in a state of temporary debugging stupidity :)

**The problem**
This code, in this scenario, causes Edge/Chrome to hang indefinitely. The promise's catch handler is never called.

```
return $("div>")
    .addClass("error")
    .append(
        $("<span>").addClass("bx bx-error"),
        $element,
        $("<br/>"),
        $("<span>").text("[Pending Deletion]")
    )
;
```

The fix:

```
return $("<div>")
    .addClass("error")
    .append(
        $("<span>").addClass("bx bx-error"),
        $element,
        $("<br/>"),
        $("<span>").text("[Pending Deletion]")
    )
;
```

Firefox correctly invokes the corresponding promise's catch clause and displays the error.
